sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -23,6 +23,8 @@
 from kfp_tekton.compiler._op_to_template import _get_base_step, _add_mount_path, _prepend_steps
 from os import environ as env
 
+BIG_DATA_MIDPATH = "artifacts/$(context.pipelineRun.uid)"
```

Comment:
I also updated the KFP API server above to replace all the `$(context.pipelineRun.uid)` to KFP Run ID, it includes all the `$(context.pipelineRun.uid)` in the sub-pipeline spec

Reply:
I have few questions on the requirements:
1. Are you going to refer/produce the artifacts inside a recursion graph? 
2. If we need to refer a variable from the Tekton labels, we need to convert that into an environment variable inside the container. In that case, we will need to pass the artifact path as a bash command arguments. It should be automatically mapped to a bash command argument if you use `OutputPath` in the python function. So I want to make sure there's no other edge cases when you use the big data passing.

I can add the original pipelineRun name instead of RunID, just need to understand the requirements better so it can cover the edge cases.

Reply:
1. I don't think it's needed for now, no.
2. Do you mean it will only work for bash-based components? That would be bad news, as it would heavily restrict the user-defined components, wouldn't it?

Reply:
@Udiknedormin 
1. That's great
2. Only the entrypoint command has to be bash/sh based. You can use bash to run python program like `sh -c python program.py --path $SOME_ENV`. Here is how KFP wrap any program with the sh entrypoint. 
https://github.com/kubeflow/kfp-tekton/blob/55245a20c3bec945fe02feb16f5c7dc5d240ef3f/sdk/python/tests/compiler/testdata/big_data_passing.yaml#L53-L90

Reply:
I will update the PR to use the original pipelineRun name by populating them as environment variable.